numpyro/distributions/mixtures.py

```diff
@@ -145,7 +144,7 @@ def sample_with_intermediates(self, key, sample_shape=()):
     def sample(self, key, sample_shape=()):
         return self.sample_with_intermediates(key=key, sample_shape=sample_shape)[0]
 
-    @validate_sample
+    # @validate_sample
```

Comment:
I think we need to enable `validate_sample` for all components. One way to achieve this is to enable it in the constructor
```
for d in component_distributions:
  d._validate_args = True
```

Reply:
Oops, sorry. You already addressed this in `component_log_probs`.